src/esp/core/managedContainers/ManagedFileBasedContainer.h

```diff
@@ -136,13 +136,13 @@ class ManagedFileBasedContainer : public ManagedContainer<T, Access> {
    * no appropriate specialization exists for passed type of document.
    *
    * @tparam type of document
-   * @param filename name of potentia document to load
+   * @param filename name of potential document to load
    * @param resDoc a reference to the document to be parsed.
    * @return whether document has been loaded successfully or not
    */
   template <class U>
   bool verifyLoadDocument(const std::string& filename,
-                          CORRADE_UNUSED U& resDoc) {
+                          CORRADE_UNUSED std::shared_ptr<U>& resDoc) {
```

Comment:
Is wrapping them as shared_ptrs really the right move? There is additional overhead here vs a raw ptr.

Reply:
It's an out-variable - the document is created in the verifyLoadDocument function, in temporary space. A regular pointer is going to choke after that function closes.

Reply:
How about a unique ptr?

Reply:
Yeah, that's only a little overhead over a non-smart ptr.